test/test_sparse.py

```diff
@@ -82,6 +84,59 @@ def ignore_op(func):
             torch.ops.aten._values.default,
         )
 
+class TestSparse000(TestCase):
+    # we want to run the tests below once before all other test cases,
+    # using 000 ensures that
+
+    # this test is disabled temporarily
+    def test_legacy_warnings(self):
+
+        def f1():
+            "UserWarning: torch.sparse.SparseTensor() is deprecated."\
+                "  Please use torch.sparse_coo_tensor((0,), dtype=)"
+            x_ref = torch.sparse_coo_tensor((0,), dtype=torch.float64)
+            x = torch.sparse.DoubleTensor()
+            self.assertEqual(x, x_ref)
+
+        def f2():
+            "UserWarning: torch.sparse.SparseTensor(cdata=x._cdata) is deprecated."\
+                "  Please use torch.sparse_coo_tensor(x._indices(), x._values(), x.shape)"
+            x_ref = torch.tensor([[1, 2], [3, 4]], dtype=torch.float64).to_sparse()
+            x = torch.sparse.DoubleTensor(cdata=x_ref._cdata)
+            y = torch.sparse_coo_tensor(x._indices(), x._values(), x.shape)
+            self.assertEqual(x, x_ref)
+            self.assertEqual(y, x_ref)
+
+        def f3():
+            "UserWarning: torch.sparse.SparseTensor(indices, values, *, device=) is deprecated."\
+                "  Please use torch.sparse_coo_tensor(indices, values, dtype=, device=)"
+            x_ref = torch.sparse_coo_tensor([[0, 0, 1, 1], [0, 1, 0, 1]], [1, 2, 3, 4], dtype=torch.float64)
+            x = torch.sparse.DoubleTensor(torch.tensor([[0, 0, 1, 1], [0, 1, 0, 1]]),
+                                          torch.tensor([1, 2, 3, 4], dtype=torch.float64))
+            self.assertEqual(x, x_ref)
+
+        def f4():
+            "UserWarning: torch.sparse.SparseTensor(indices, values, shape, *, device=) is deprecated."\
+                "  Please use torch.sparse_coo_tensor(indices, values, shape, dtype=, device=)"
+            x_ref = torch.sparse_coo_tensor([[0, 0, 1, 1], [0, 1, 0, 1]], [1, 2, 3, 4], (2, 3), dtype=torch.float64)
+            x = torch.sparse.DoubleTensor(torch.tensor([[0, 0, 1, 1], [0, 1, 0, 1]]),
+                                          torch.tensor([1, 2, 3, 4], dtype=torch.float64), (2, 3))
+            self.assertEqual(x, x_ref)
+
+        def f5():
+            "UserWarning: torch.sparse.SparseTensor(shape, *, device=) is deprecated."\
+                "  Please use torch.sparse_coo_tensor(shape, dtype=, device=)"
+            x_ref = torch.sparse_coo_tensor((2, 3), dtype=torch.float64)
+            x = torch.sparse.DoubleTensor(2, 3)
+            self.assertEqual(x, x_ref)
+
+        for test_f in [f1, f2, f3, f4, f5]:
+            cerr = io.StringIO()
+            with redirect_stderr(cerr):
+                test_f()
+                test_f()  # checks warns once
+            FileCheck().check_count(str=test_f.__doc__, count=1, exactly=True).run(cerr.getvalue())
```

Comment:
You might be able to use [`assertWarns`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertWarns).

Reply:
yes, thanks for the hint!

Btw, we have `assertWarnsOnceRegex` in torch/testing/_internal/common_utils.py that naming is misleading: it calls `set_warn_always(True)` to force `TORCH_WARN_ONCE` to become `TORCH_WARN` but that behavior is contradictory to the method name --- I would expect that it would implement a true test for the "warn once" feature.